sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -408,6 +408,7 @@ abstract class NewRule {
      *
      * Each section must have a different section key when they are non-contextual.
      * As soon as one section is contextual for a section key, all sections with that key must be contextual. The pair "section key, context key" pair must still be unique.
+     * If several sections provide contexts, the provided context keys must be shared between sections.
```

Comment:
Nitpick: I would replace _shared_ with _the same_ as you used in the error message

Reply:
I agree, thanks for the suggestion